### DISCUSSION TITLE
v1.2.0-rc1

### DISCUSSION THREAD

**Discussion by mosteo** [alire-project/alire#1026] [Announcements]
## Main new features
* New `alr exec` command.
* Custom command aliases.
* Build profiles.
* Monorepos.

See https://github.com/alire-project/alire/blob/master/doc/user-changes.md for details.

## What's Changed
* Incremental building of dependencies by @mosteo in https://github.com/alire-project/alire/pull/832
* Fix missing action triggers, again by @mosteo in https://github.com/alire-project/alire/pull/835
* Refactoring: remove string subprograms from Alire.Utils/Alr.Utils and use AAA.Strings by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/837
* Alire.Config: replace with CLIC.Config by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/841
* Port to `ada_toml` for TOML 1.0 by @mosteo in https://github.com/alire-project/alire/pull/836
* Alr.Commands.Build: forward subcommand args to gprbuild by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/850
* Revamped `alr version` output by @mosteo in https://github.com/alire-project/alire/pull/852
* Update CLIC dependency to avoid build failure on older GNAT versions by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/872
* Enable and update self-build by @mosteo in https://github.com/alire-project/alire/pull/798
* Use Stopwatch for elapsed time image (#868) by @jquorning in https://github.com/alire-project/alire/pull/870
* Reorder switches for alr test subprocess in Default_Test by @jquorning in https://github.com/alire-project/alire/pull/871
* New command exec and support of aliases by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/853
* Alire.Releases: fix dependencies in YAML output by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/867
* Alr.Commands.Init: fix .gitignore for bin crates by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/877
* Test case for bug #875 (virtual crate environment) by @mosteo in https://github.com/alire-project/alire/pull/878
* Don't err when unsetting a non-existing config key by @mosteo in https://github.com/alire-project/alire/pull/884
* Alire.Crate_Configuration: Generate Pure and Preelaborate Ada package by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/894
* Build profiles and switches system by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/895
* Build profiles and swtiches continued by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/896
* Fix space typo in trace for unknown distribution by @jquorning in https://github.com/alire-project/alire/pull/901
* Replace invalid option --skip-compile with --skip-build in alr clean by @onox in https://github.com/alire-project/alire/pull/903
* Bugfix for nested empty tables in manifest by @mosteo in https://github.com/alire-project/alire/pull/906
* Fix identifier typo. by @pyjarrett in https://github.com/alire-project/alire/pull/908
* Fix spelling of  alr by @bracke in https://github.com/alire-project/alire/pull/909
* Rename to GNAT_Suppress_Runtime_Check by @jquorning in https://github.com/alire-project/alire/pull/913
* Rename to Make_Switches_Map by @jquorning in https://github.com/alire-project/alire/pull/914
* Rename Classifiable interface by @jquorning in https://github.com/alire-project/alire/pull/916
* Fix several typos in comments by @jquorning in https://github.com/alire-project/alire/pull/915
* Fix several typos in trace and output by @jquorning in https://github.com/alire-project/alire/pull/912
* Bump clic to fix infinite loop on multichoice on macOS by @mosteo in https://github.com/alire-project/alire/pull/927
* Don't attempt to build a binary release by @mosteo in https://github.com/alire-project/alire/pull/928
* alr toolchain: fix gprbuild typo in help by @Chris660 in https://github.com/alire-project/alire/pull/929
* Refactor scattered platform code into Alire.Platforms.Current by @mosteo in https://github.com/alire-project/alire/pull/931
* New config option to disable community index by @mosteo in https://github.com/alire-project/alire/pull/933
* Log command line before parsing & execution by @mosteo in https://github.com/alire-project/alire/pull/938
* alr.py: ensure baseline git configuration in tests by @mosteo in https://github.com/alire-project/alire/pull/936
* Display pins in show/with commands by @mosteo in https://github.com/alire-project/alire/pull/940
* doc: Add an introduction and update getting-started.md by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/899
* Support for publishing of crates in subdirectories of repositories by @mosteo in https://github.com/alire-project/alire/pull/939
* Emit all found executables by @jquorning in https://github.com/alire-project/alire/pull/946
* Remove pins from manifest during `alr publish` by @mosteo in https://github.com/alire-project/alire/pull/947
* Leftover comments from platform refactor by @mosteo in https://github.com/alire-project/alire/pull/950
* New property for detected host architecture by @mosteo in https://github.com/alire-project/alire/pull/932
* Alire.Properties.Build_Switches: Add support for Ada version switches by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/953
* Alire.Environment: export <CRATE_NAME>_PREFIX for all crates of solution by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/954
* Change default GPRbuild switches by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/955
* Alr.Commands.Exec: -P switch to add project file on the command line by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/956
* alire.toml: bump gnatcoll-slim commit to match git submodule by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/959
* Alire.Environment: change _PREFIX to _ALIRE_PREFIX by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/958
* Alire.Crate_Configuration: Add a Crate_Name constant for all configs by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/957
* Silence git when not running in a tty by @mosteo in https://github.com/alire-project/alire/pull/963
* Remove git --progress flag (fixes #966) by @JeremyGrosser in https://github.com/alire-project/alire/pull/967
* Alire.Crate_Configuration: add built-in values for host OS, Arch and Distro by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/961
* Remapping of ARM64 to AARCH64 by @mosteo in https://github.com/alire-project/alire/pull/968
* Update msys2 installer and allow user to set version by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/970
* Alire.Manifest: Remove "Added by alr" comments by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/977
* Use proper index versioning for backwards compatibility by @mosteo in https://github.com/alire-project/alire/pull/975
* Fix some rich output that eluded --no-color/--no-tty by @mosteo in https://github.com/alire-project/alire/pull/978
* Reject manifest keys with underscores by @mosteo in https://github.com/alire-project/alire/pull/980
* Alr.Commands.Init: Fix missing config src dir by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/981
* Alire.Utils.Switches: Disable contracts for Development profile by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/979
* Alr.Command: fix gnatcov built-in alias by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/986
* Alire.Crate_Configuration: Fix pragma No_Elaboration_Code by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/987
* New `action` subcommand by @mosteo in https://github.com/alire-project/alire/pull/983
* Refactor Alire.Features.Index to Alire.Index_On_Disk.Loading by @mosteo in https://github.com/alire-project/alire/pull/992
* Alire.Properties.Configurations: revert forced mixed case by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/998
* Update getting-started.md by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/1000
* Alr.Commands.Init: change default version for new crates to 0.1.0-dev by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/1004
* Alr.Commands.Init: Create share folder for resources by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/1003
* Implement load of individual crates from index by @mosteo in https://github.com/alire-project/alire/pull/1006
* Cache index detection and crate loading by @mosteo in https://github.com/alire-project/alire/pull/1008
* Alire.Index_On_Disk: Drop index cache after deletion by @mosteo in https://github.com/alire-project/alire/pull/1010
* Bump index version by @mosteo in https://github.com/alire-project/alire/pull/1009
* Show just the version with `alr --version` by @mosteo in https://github.com/alire-project/alire/pull/1013
* Bump msys installer in devel branch by @mosteo in https://github.com/alire-project/alire/pull/1017
* tests/crate_config/basic: fix arm64 expected host by @Fabien-Chouteau in https://github.com/alire-project/alire/pull/1015
* Load the index when dependencies are needed by @mosteo in https://github.com/alire-project/alire/pull/1019
* Properly detect domain in publishing assistant by @mosteo in https://github.com/alire-project/alire/pull/1020
* Don't try to detect external tools when they're from binary archives by @mosteo in https://github.com/alire-project/alire/pull/1018
* Create zipball with full sources for releases by @mosteo in https://github.com/alire-project/alire/pull/1023
* New `alr get --dirname` switch by @mosteo in https://github.com/alire-project/alire/pull/1014
* Fix retrieval of git root and prefix inside repo by @mosteo in https://github.com/alire-project/alire/pull/1024

## New Contributors
* @bracke made their first contribution in https://github.com/alire-project/alire/pull/909
* @Chris660 made their first contribution in https://github.com/alire-project/alire/pull/929
* @JeremyGrosser made their first contribution in https://github.com/alire-project/alire/pull/967

**Full Changelog**: https://github.com/alire-project/alire/compare/v1.1.2...v1.2.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/alire-project/alire/releases/tag/v1.2.0-rc1'>v1.2.0-rc1</a>.</em>

**Comment by Joebeazelman** [comment]
The long list of features pushed the release all the way to the very bottom, making it easy to overlook.  You might want to add a direct link to the latest release somewhere in the introduction of both the readme file and the Alire site. For some reason, GItHub also collapses the download area and calls it  "Assets", making it even harder to find.

**Reply by mosteo** [reply]
Thanks for the observation, @Joebeazelman, I will take it into account for future releases. Unfortunately it seems that the "assets" section can not be directly referenced?

**Reply by Joebeazelman** [reply]
Not sure what you're asking, but it's in the release section of the latest version. You have to scroll towards the bottom and you'll see it marked as "Assets" or there will be a palette with a bunch of downloadable links.

**Reply by mosteo** [reply]
I mean that I don't see a way to provide a link to that "Assets" area direcly, to make it more evident. Otherwise I'm not understanding your 

> The long list of features pushed the release all the way to the very bottom, making it easy to overlook.

In my case the release does show at the top, so now I'm now sure what you mean.
